src/hooks/useScrollShadow.js

```diff
@@ -0,0 +1,94 @@
+// very difficult to test with JSDom, some basic interaction is tested in ScrollableContainer
+/* istanbul ignore file */
+import { useEffect } from 'react'
+import throttle from 'lodash.throttle'
+
+const BOX_SHADOW_INITIAL = '0 0 0 1px rgba(0, 0, 0, 0.1)'
+const BOX_SHADOW_SCROLLED =
+  '0 0 0 1px rgba(0, 0, 0, 0.1), 0 0 0 5px rgba(0, 0, 0, 0.05)'
+
+export default function useScrollShadow({
+  bottomRef,
+  scrollableRef,
+  topRef,
+  shadows = {},
+}) {
+  const initialShadow = shadows.initial || BOX_SHADOW_INITIAL
+  const scrolledShadow = shadows.scrolled || BOX_SHADOW_SCROLLED
+
+  useEffect(() => {
+    setTimeout(() => {
+      if (scrollableRef && scrollableRef.current != null) {
+        const topElement = topRef && topRef.current
+        const bottomElement = bottomRef && bottomRef.current
+
+        if (topElement) {
+          topElement.style.boxShadow = 'var(--scroll-shadow)'
+        }
+
+        if (bottomElement) {
+          bottomElement.style.boxShadow = 'var(--scroll-shadow)'
+        }
+
+        const { isBottomScrolled } = handleShadows(scrollableRef)
+
+        if (isBottomScrolled) {
+          bottomElement.style.setProperty('--scroll-shadow', scrolledShadow)
+        }
+      }
+    }, 100)
+  }, [scrollableRef, bottomRef, topRef, scrolledShadow])
```

Comment:
I think the refs here are not needed. Since they are not mutable from react perspective, the effect would never be called because the ref has changed, so there's no need to put them as dependencies. At least, that's what linter is telling me in hs-app-ui :smile:

Reply:
I see, I kind of thought the same but the ESlint rule about dependencies kept nagging me to put them, I'll just add an ignore directive

Reply:
If it's bothering, then maybe I am wrong :smile: But I am sure it was nagging me the other way around. What about when you do `scrollableRef.current` instead?

Reply:
what you could do is extract the current value and used that as a dependency

```javascript
const {current: scrollableRefElement} = scrollableRef

useEffect(() = > {
....
}, [scrollableRefElement, bottomRef, topRef, scrolledShadow]))
```

Reply:
The hook supports passing DOM elements directly too now, so I added the dependencies